.htaccess

```
@@ -145,22 +147,20 @@ Redirect "/bookmarklet" "/products/bookmarklet"
 # special case because of specific filenames : these don't redirect properly otherwise
 
 # Rewrite nul -> _nul, index -> _index
-RewriteRule "^developer/language/reference/(nul|index)(\.php)?$" "developer/language/reference/_$1.php" [L]
+RewriteRule "^developer/language/reference/(nul|index)(\.php)?$" "developer/language/reference/_$1.php" [END]
 
 # Remove index or index.php and redirect (and stop processing)
-RewriteCond "$1/index" -f [OR]
-RewriteCond "$1/index.php" -f
 RewriteRule "^((.+)/)?index(\.php)?$" "/$1" [R,L]
 
 # Remove .php extension and redirect
-RewriteCond "$1.php" -f
+RewriteCond "%{DOCUMENT_ROOT}/$1.php" -f
```

> Also, the keyboard php extension isn't getting removed, so the redirect is loading something like:
http://localhost:8055/keyboard/georgian/1.0/georgian.php

Is this the reason? (We now expect a top-level file whereas previously it could be in a subdirectory)

---

```suggestion
RewriteCond "$1.php" -f
```

As @ermshiperete says, I think this should be restored to the original.

---

Done.

But the keyboard pages still show `.php` :disappointed:

---

Ping me if you want to rubber-ducky on this next week?